File: MORM.CSharp/MORM.CSharp/Classes/Comando.cs
using System;
using System.Linq;

namespace MORM.CSharp.Classes
{
    public static class Comando
    {
        //-- tipo database

        public static TipoDatabase TipoDatabase { get; set; } //= TipoDatabase.Oracle;
        
        //-- tabela
        
        public static Tabela GetTabela(this Type type)        
        {
        	if (type == typeof(CollectionItem))
            {
                var collection = Activator.CreateInstance(type) as CollectionItem;
                return collection.GetTabela();
            }
            else
                return null;
        }
        
        //-- campos

        public static Campos GetCampos(this Type type)
        {
        	if (type == typeof(CollectionItem))
            {
                var collection = Activator.CreateInstance(type) as CollectionItem;
                return collection.GetCampos();
            }
            else
                return null;
        }

        //-- value

        private static string GetValueStr(this object obj, string atributo)
        {
            var value = obj.GetType().GetProperties()
            	.FirstOrDefault(x => x.Name == atributo)
            	.GetValue(obj);
            
            return (
                value == null ? "null" :
            	value is bool ? "'" + ((bool)value ? "T" : "F") + "'" :
            	value is DateTime ? TipoDatabase.GetValueData((DateTime)value) :
            	value is decimal ? value.ToString().Replace(",", ".") :
            	value is double ? value.ToString().Replace(",", ".") :
            	value is int ? value.ToString() :
            	value is string ? "'" + value.ToString().Replace("'", "''") + "'" : value.ToString());
        }

        //-- string

        public static void AddString(ref string str, string val, string sep, string ini = "")
        {
            str += (!string.IsNullOrWhiteSpace(str) ? sep : ini) + val;
        }

        //-- select

        public static string GetSelect(this Type type, string where = "")
        {
            var tabela = type.GetTabela();
            var campos = type.GetCampos();
            var fieldsAtr = string.Empty;
            var fields = string.Empty;

            foreach (var campo in campos)
            {
                AddString(ref fieldsAtr, campo.Atributo + " as \"" + campo.Atributo + "\"", ", ");
                AddString(ref fields, campo.Nome + " as " + campo.Atributo, ", ");
            }

            return 
                "select " + fieldsAtr + 
            	" from (select " + fields + " from " + tabela.Nome + ")" +
                (!string.IsNullOrWhiteSpace(where) ? " where " + where : "") ;
        }

        public static string GetSelect(this object obj)
        {
            var campos = obj.GetType().GetCampos();
            var where = string.Empty;

            foreach (var campo in campos)
                if (campo.Tipo == CampoTipo.Key)
                    AddString(ref where, campo.Atributo + " = " + obj.GetValueStr(campo.Atributo), " and ");

            return GetSelect(obj.GetType(), where);
        }

        //-- insert

        public static string GetInsert(this object obj)
        {
            var tabela = obj.GetType().GetTabela();
            var campos = obj.GetType().GetCampos();
            var fields = string.Empty;
            var values = string.Empty;

            foreach (var campo in campos)
            {
                AddString(ref fields, campo.Nome, ", ");
                AddString(ref values, obj.GetValueStr(campo.Atributo), ", ");
            }

            return 
                "insert into " + tabela.Nome + 
            	" (" + fields + 
            	") values (" + values + ")";
        }

        //-- update

        public static string GetUpdate(this object obj)
        {
            var tabela = obj.GetType().GetTabela();
            var campos = obj.GetType().GetCampos();
            var sets = string.Empty;
            var where = string.Empty;

            foreach (var campo in campos)
                if (campo.Tipo == CampoTipo.Key)
                    AddString(ref where, campo.Nome + " = " + obj.GetValueStr(campo.Atributo), " and ");
                else
                    AddString(ref sets, campo.Nome + " = " + obj.GetValueStr(campo.Atributo), ", ");

            return
                "update " + tabela.Nome + 
            	" set " + sets + 
            	" where " + where;
        }

        //-- delete

        public static string GetDelete(this object obj)
        {
            var tabela = obj.GetType().GetTabela();
            var campos = obj.GetType().GetCampos();
            var where = string.Empty;

            foreach (var campo in campos)
                if (campo.Tipo == CampoTipo.Key)
                    AddString(ref where, campo.Nome + " = " + obj.GetValueStr(campo.Atributo), " and ");

            return
                "delete from " + tabela.Nome + 
            	" where " + where;
        }
    }
}

File: MORM.Java/src/MORM/Java/Classes/Comando.java
package MORM.Java.Classes;

import MORM.Java.Classes.Mapping.Campo;
import MORM.Java.Classes.Mapping.CampoTipo;
import MORM.Java.Classes.Mapping.Campos;
import MORM.Java.Classes.Mapping.Tabela;

public class Comando
{
	public Comando(TipoDatabase tipoDatabase)
	{
		TipoDatabase = tipoDatabase;
	}
	
	public TipoDatabase TipoDatabase;
	
    protected Tabela GetTabela(Class<? extends CollectionItem> collectionItemClass)
    {
    	Tabela tabela = null;
		try {
	  		CollectionItem collecitonItem = collectionItemClass.newInstance();
	  		tabela = collecitonItem.GetTabela();
		} catch (InstantiationException e) {
			e.printStackTrace();
		} catch (IllegalAccessException e) {
			e.printStackTrace();
		}
  		return tabela;
    }
    
    protected Campos GetCampos(Class<? extends CollectionItem> collectionItemClass)
    {
    	Campos campos = null;
		try {
	  		CollectionItem collecitonItem = collectionItemClass.newInstance();
	  		campos = collecitonItem.GetCampos();
		} catch (InstantiationException e) {
			e.printStackTrace();
		} catch (IllegalAccessException e) {
			e.printStackTrace();
		}
  		return campos;
    }
    
    protected void AddString(StringBuilder str, String sub, String sep, String ini)
    {
    	str.append((str.length() > -1 ? sep : ini) + sub);
    }
    
    private String GetValueStr(CollectionItem collectionItem, String atributo) 
    {
		return null;
	}
	
	public String GetSelect(Class<? extends CollectionItem> collectionItemClass, String where)
    {
        Tabela tabela = GetTabela(collectionItemClass);
        Campos campos = GetCampos(collectionItemClass);
        StringBuilder fieldsAtr = new StringBuilder();
        StringBuilder fields = new StringBuilder();

        for (Campo campo : campos)
        {
            AddString(fieldsAtr, campo.Atributo + " as \"" + campo.Atributo + "\"", ", ", "");
            AddString(fields, campo.Campo + " as " + campo.Atributo, ", ", "");
        }

        return
            "select " + fieldsAtr.toString() +
            " from (" +
              "select " + fields.toString() +
              " from " + tabela.Nome + 
            ")" + (where != "" ? " where " + where : "");
    }
    
    public String GetSelect(CollectionItem collectionItem)
    {
        Campos campos = collectionItem.GetCampos();
        StringBuilder where = new StringBuilder();

        for (Campo campo : campos)
            if (campo.Tipo == CampoTipo.Key)
                AddString(where, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), " and ", "");

        return GetSelect(collectionItem.getClass(), where.toString());
    }

	public String GetInsert(CollectionItem collectionItem)
    {
        Tabela tabela = collectionItem.GetTabela();
        Campos campos = collectionItem.GetCampos();
        StringBuilder fields = new StringBuilder();
        StringBuilder values = new StringBuilder();
        
        for (Campo campo : campos)
        {
            AddString(fields, campo.Atributo, ", ", "");
            AddString(values, GetValueStr(collectionItem, campo.Atributo), ", ", "");        
        }

        return
            "insert into " + tabela.Nome + 
            "(" + fields.toString() + 
            ") values (" + values + ")" ;
    }
    
    public String GetUpdate(CollectionItem collectionItem)
    {
        Tabela tabela = collectionItem.GetTabela();
        Campos campos = collectionItem.GetCampos();
        StringBuilder sets = new StringBuilder();
        StringBuilder where = new StringBuilder();

        for (Campo campo : campos)
            if (campo.Tipo == CampoTipo.Key)
                AddString(sets, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), " and ", "");
            else
                AddString(sets, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), ", ", "");

        return
            "update " + tabela.Nome + 
            " set " + sets.toString() + 
            " where " + where.toString();
    }
    
    public String GetDelete(CollectionItem collectionItem)
    {
        Tabela tabela = collectionItem.GetTabela();
        Campos campos = collectionItem.GetCampos();
        StringBuilder where = new StringBuilder();
        
        for (Campo campo : campos)
            if (campo.Tipo == CampoTipo.Key)
                AddString(where, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), " and ", "");
        
        return
            "delete from " + tabela.Nome + 
            " where " + where.toString();
    }
}